plugins/catalog-backend/src/service/DefaultLocationService.ts

```diff
@@ -98,6 +98,14 @@ export class DefaultLocationService implements LocationService {
       });
 
       if (processed.ok) {
+        const { metadata, kind } = processed.completedEntity;
+        if (
+          entities.some(
+            e => e.metadata.name === metadata.name && e.kind === kind,
```

Comment:
Even though it rarely changes, we should check namespace here as well just to be sure.